### PR TITLE
Feat(utils): improve web-component interopt

### PR DESCRIPTION
**Pull request by odinr**
# Component Development

improves interopt between react and web-components

### Type

- [ ] Feature
- [x] Fix
- [x] Hotfix (should release ASAP)
- [ ] Maintenance (deps only)

#### Create User Story(DevOps) or Issue(Github)

- [ ] Describe functional requirements
- [ ] Design plan/suggestion (e.g. images, link to eds design or relevant figma links, mention of eds tokens to use)
- [ ] Communication with UX *(evaluate if needed)*

#### Development

- [ ] Describe the component
- [ ] Code funtional requirements
- [ ] Create stories
- [ ] Create snapshots *(evaluate if needed)*
- [ ] Remove unused imports and commented code
- [ ] Lint code with EsLint (and follow best practises for [code quality](https://docs.fusion-dev.net/development/frontend/code-quality/)
  
#### Create Pull Request

- [ ] Assign relevant reviewers in Github and Cromatic
- [ ] Inform colleagues of pending reviews. We use Teams for this purpose, under fusion-frontend team please tag involved reviewers in the [Code review channel](https://teams.microsoft.com/l/channel/19%3a056a9ba8d8d84a058b24762f85c603ae%40thread.tacv2/Code%2520%2520review?groupId=9589ba16-68a5-47e6-97d7-7d0be0e0d3cb&tenantId=3aa4a235-b6e2-48d5-9195-7fcf05b459b0)

#### Review of Pull Request

- [ ] Follow up on comments from code review and implement necessary changes *(evaluate if needed)*
- [ ] Review approved in github (and chromatic)

#### Publish Code

- [ ] Increment version number
- [ ] Merge PR
- [ ] Consider letting fusion developers know this this shiny new thing exists

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,12 +5,6 @@
   "type": "module",
   "description": "monorepo of fusion react components",
   "main": "index.js",
-  "engines": {
-    "node": "^16 || ^18 || ^20",
-    "npm": "please-use-pnpm",
-    "yarn": "please-use-pnpm",
-    "pnpm": ">=9"
-  },
```
Comment (asbjornhaland) [inline package.json:13]: Please don't remove this.

File: packages/utils/src/element-attributes.ts [hunk 2]
```diff
@@ -28,7 +28,6 @@ export const elementAttributes = <T extends Partial<Record<keyof T, unknown>> =
       default:
         return value ? Object.assign(cur, { [key]: value }) : cur;
     }
-    return cur;
```
Comment (asbjornhaland) [inline packages/utils/src/element-attributes.ts:31]: Why is this removed?

File: packages/utils/src/create-element.ts [hunk 3]
```diff
@@ -79,6 +80,7 @@ export const createComponent = <E extends HTMLElement, P extends Record<string,
     /** properties which React should handle */
     const reactProps = useMemo(() => {
       const entries = Object.entries(props || {});
+      console.log(entries);
```
Comment (asbjornhaland) [inline packages/utils/src/create-element.ts:83]: please remove console.log.

File: packages/utils/src/create-element.ts [hunk 4]
```diff
@@ -100,6 +102,8 @@ export const createComponent = <E extends HTMLElement, P extends Record<string,
       );
     }, [ref, props]);
 
+    console.log('reactProps', reactProps);
```
Comment (asbjornhaland) [inline packages/utils/src/create-element.ts:105]: please remove console.log.

**Review by asbjornhaland — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (eikeland) [reply package.json:13]: we should also add node 22 there

File: packages/utils/src/element-attributes.ts (hunk 2 above)
Comment (eikeland) [reply packages/utils/src/element-attributes.ts:31]: since the switch has a default i guess.
